apps/passport-client/components/screens/ScannedTicketScreens/PodboxScannedTicketScreen/sections/PodboxInstantCheckin.tsx

```diff
@@ -0,0 +1,72 @@
+import { Spacer } from "@pcd/passport-ui";
+import { useEffect } from "react";
+import { RippleLoader } from "../../../../core/RippleLoader";
+import { ErrorContainer, StatusContainer } from "../PodboxScannedTicketScreen";
+import { PodboxTicketActionErrorSection } from "./actions/PodboxTicketErrors";
+import { useExecuteTicketAction } from "./actions/useExecuteTicketAction";
+
+export function InstantCheckin({
+  ticketId,
+  eventId
+}: {
+  ticketId: string;
+  eventId: string;
+}): JSX.Element {
+  const executor = useExecuteTicketAction({
+    eventId,
+    ticketId,
+    action: {
+      checkin: true
+    }
+  });
+
+  useEffect(() => {
+    executor.execute();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  if (executor.loading) {
+    return (
+      <>
+        <RippleLoader />
+        <Spacer h={8} />
+      </>
+    );
+  }
+
+  if (executor.result?.success) {
+    const value = executor.result.value;
+    if (value.success) {
+      // For the richer "action executor" system used when camera-scanning, we
+      // have different success messages. These are fetched from the pre-check
+      // API, which we do not use here. However, since we can only do check-in,
+      // it's OK that the message is hard-coded.
+      const successMessage = "Checked In";
```

Comment:
note that this should be "Swag Given" for the swag scanners - is there a good client-side way to fetch if a user has a superuser ticket vs. a userPermission ticket? cc @ichub

Reply:
https://github.com/proofcarryingdata/zupass/blob/ivan/swag-stats/apps/passport-client/components/screens/ScannedTicketScreens/PodboxScannedTicketScreen/sections/actions/checkin/CheckInActionSection.tsx#L79

> is there a good client-side way to fetch if a user has a superuser ticket vs. a userPermission ticket?

not specifically this, but there's a way to get the 'success message' copy from the `precheck` value as linked

Reply:
will be done in a follow-up PR